modules/monitoring-resizing-a-persistent-volume.adoc

```diff
@@ -0,0 +1,145 @@
+----
+
+.. Add a new storage size for the PVC configuration for the component under `data/config.yaml`:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    <component>: #<1>
+      volumeClaimTemplate:
+        spec:
+          resources:
+            requests:
+              storage: <amount_of_storage> #<2>
+----
+<1> The component for which you want to change the storage size.
+<2> Specify the new size for the storage volume. It must be greater that the previous value.
++
+The following example sets the new PVC request to 100 gigabytes for the Prometheus instance that monitors core {product-title} components:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      volumeClaimTemplate:
+        spec:
+          resources:
+            requests:
+              storage: 100Gi
+----
+
+** *If you are configuring components that monitor user-defined projects*:
++
+[NOTE]
+====
+You can resize the volumes for the Thanos Ruler and Prometheus instances that monitor user-defined projects.
+====
++
+.. Edit the `user-workload-monitoring-config` `ConfigMap` object in the `openshift-user-workload-monitoring` project:
++
+[source,terminal]
+----
+$ oc -n openshift-user-workload-monitoring edit configmap user-workload-monitoring-config
+----
+
+.. Update the PVC configuration for the monitoring component under `data/config.yaml`:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: user-workload-monitoring-config
+  namespace: openshift-user-workload-monitoring
+data:
+  config.yaml: |
+    <component>: #<1>
+      volumeClaimTemplate:
+        spec:
+          resources:
+            requests:
+              storage: <amount_of_storage> #<2>
+----
+<1> The component for which you want to change the storage size.
+<2> Specify the new size for the storage volume. It must be greater that the previous value.
++
+The following example sets the new PVC request to 20 gigabytes for Thanos Ruler:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: user-workload-monitoring-config
+  namespace: openshift-user-workload-monitoring
+data:
+  config.yaml: |
+    thanosRuler:
+      volumeClaimTemplate:
+        spec:
+          resources:
+            requests:
+              storage: 20Gi
+----
++
+[NOTE]
+====
+Storage requirements for the `thanosRuler` component depend on the number of rules that are evaluated and how many samples each rule generates.
+====
+
+. Save the file to apply the changes.
++
+[WARNING]
+====
+When you save changes to a monitoring config map, the pods and other resources in the related project might be redeployed. The monitoring processes running in that project might also be restarted.
```

Comment:
Is it a "might" or a "will"? If the pods are always rolled out and it leads to a (temporary) outage, it must be clearly communicated.

Reply:
@simonpasquier Thank you for this comment! We have a similar note throughout the docs for many different procedures (just in the "configuring the monitoring stack" section I counted 17 similar notes) and I wondered about this too actually. But because the note is so common, I did not question it further as I assumed it was already discussed in the past.

@juzhao bacause we have this note on so many places in the docs, I wanted to check with you how accurate this actually is. Is it different for different procedures or are certain pods and resources always redeployed if we make any changes to a monitoring config map? I would create an appropriate Jira issue for this based on the answer :+1:

Reply:
@eromanova97 any saved changes in the configmap would always make the pods recreated, change to `will` is correct

Reply:
Great, thank you for the confirmation! I will proceed to fix it here and create an issue to fix it in the other places as well :+1:

Reply:
Done (I just used the present tense as I need to avoid using future in our docs :+1: )

Reply:
There's a difference though between
* the statefulset is updated and the pods are rolled out gradually (without service interruption).
* the statefulset is recreated (as in this PVC procedure) meaning that for some time, we have no monitoring at all.

Reply:
@simonpasquier because I created a new issue for this: https://issues.redhat.com/browse/OBSDOCS-1155
I hope it is okay with you to move this PR forward for now (I don't want to block it anymore). We can resolve the issue above in this new Jira. Thank you!

Reply:
works for me!